### PR TITLE
Move world time and light to lua

### LINKED ISSUE #258: The length of day and night in config.lua
Just like in the topic, 
example:
dayLength = 8 --- in hour
nightLength = 4 --- in hour
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1337284-the-length-of-day-and-night-in-config-lua?utm_campaign=plugin&utm_content=tracker%2F392151&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F392151&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Move world time and light to lua

Close #258 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/scripts/creaturescripts/player/world_light_time.lua [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+event = CreatureEvent("WorldLight")
+
+function event.onLogin(player)
+    local worldLightColor, worldLightLevel = Game.getWorldLight()
+	player:sendWorldLight(worldLightColor, worldLightLevel)
+	return true
+end
+
+event:register()
```
Comment (EPuncker) [inline data/scripts/creaturescripts/player/world_light_time.lua:19]:
```suggestion
event:register()

```

File: data/scripts/globalevents/world_light.lua [hunk 2]
```diff
@@ -0,0 +1,55 @@
+local event = GlobalEvent("WorldLight")
+
+local lightConfig = {
+    day = 250,
```
Comment (EPuncker) [inline data/scripts/globalevents/world_light.lua:4]:
```suggestion
	day = 250,
```

File: data/scripts/creaturescripts/player/world_light_time.lua [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+event = CreatureEvent("WorldLight")
+
+function event.onLogin(player)
+	local worldLightColor, worldLightLevel = Game.getWorldLight()
+	player:sendWorldLight(worldLightColor, worldLightLevel)
+	return true
+end
+
+event:register()
```
Comment (ranisalt) [inline data/scripts/creaturescripts/player/world_light_time.lua:19]: Can it be packed in a single function?

File: src/luascript.cpp [hunk 4]
```diff
@@ -10931,6 +10892,35 @@ int LuaScriptInterface::luaPlayerSendCreatureSquare(lua_State* L)
...
+	Player* player = getUserdata<Player>(L, 1);
+	if (player) {
+		player->sendWorldTime(getNumber<int16_t>(L, 2));
+		pushBoolean(L, true);
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
```
Comment (ranisalt) [inline src/luascript.cpp:10923]: Get rid of those, send the packet using `NetworkMessage` from Lua instead

File: data/scripts/creaturescripts/player/world_light_time.lua (hunk 3 above)
Comment (ramon-bernardo) [reply data/scripts/creaturescripts/player/world_light_time.lua:12]: Yeah

File: src/luascript.cpp (hunk 4 above)
Comment (ramon-bernardo) [reply src/luascript.cpp]: Moved

**Review by ranisalt — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Pull request merged**